Python/traceback.c

```diff
@@ -1189,9 +1189,9 @@ dump_frame(int fd, InterpreterFrame *frame)
     }
     PUTS(fd, " in ");
 
-    if (code->co_name != NULL
-       && PyUnicode_Check(code->co_name)) {
-        _Py_DumpASCII(fd, code->co_name);
+    PyObject *name = code->co_qualname ? code->co_qualname : code->co_name;
+    if (name != NULL && PyUnicode_Check(code->co_qualname)) {
+        _Py_DumpASCII(fd, code->co_qualname);
```

Comment:
This looks wrong, you probably want to do:
```suggestion
    PyObject *name = code->co_qualname ? code->co_qualname : code->co_name;
    if (name != NULL && PyUnicode_Check(name)) {
        _Py_DumpASCII(fd, name);
```

Reply:
Yes! Good catch, thanks!